lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/SQLTranslatorUtilsTest.java

```diff
@@ -2536,7 +2539,7 @@ public void testTranslateModelWithHasLikeKeywordAndEscape() throws ParseExceptio
 		TableAndColumnMapper mapper = new TableAndColumnMapper(element, mockSchemaProvider);
 		Map<String, Object> parameters = new HashMap<>();
 		SQLTranslatorUtils.translateModel(element, parameters, userId, mapper);
-		assertEquals( "SELECT * FROM T123 WHERE ROW_ID IN ( SELECT ROW_ID_REF_C777_ FROM T123_INDEX_C777_ WHERE _C777__UNNEST IN ( :b0, :b1, :b2 ) ) AND ( ROW_ID IN ( SELECT ROW_ID_REF_C111_ FROM T123_INDEX_C111_ WHERE _C111__UNNEST LIKE :b3 ESCAPE :b5 OR _C111__UNNEST LIKE :b4 ESCAPE :b5 ) OR _C333_ = :b6 )",element.toSql());
+		assertEquals("SELECT * FROM T123 WHERE ( JSON_OVERLAPS(_C777_,JSON_ARRAY(:b0,:b1,:b2)) IS TRUE ) AND ( ( JSON_SEARCH(_C111_,'one',:b3 COLLATE 'utf8mb4_0900_ai_ci',:b5,'$[*]') IS NOT NULL OR JSON_SEARCH(_C111_,'one',:b4 COLLATE 'utf8mb4_0900_ai_ci',:b5,'$[*]') IS NOT NULL ) OR _C333_ = :b6 )",element.toSql());
 		assertEquals(1L, parameters.get("b0"));

```

Comment:
need a test for the case where the 'not has_like' -> is null

Reply:
I added the test and the query was indeed wrong, NOT HAS_LIKE(foo, 'a', 'b') was translated into JSON_SEARCH(foo, 'a') IS NULL **OR** JSON_SEARCH(foo, 'b') IS NULL. Should be instead JSON_SEARCH(foo, 'a') IS NULL **AND** JSON_SEARCH(foo, 'b') IS NULL